src/main/java/com/iridium/iridiumteams/database/IridiumUser.java

```diff
@@ -54,11 +54,19 @@ public Player getPlayer() {
         return Bukkit.getServer().getPlayer(uuid);
     }
 
-    public boolean canFly(IridiumTeams<T, ?> iridiumTeams) {
+    public boolean canFly(Optional<T> team, IridiumTeams<T, ?> iridiumTeams) {
         Player player = getPlayer();
-        if (player.hasPermission(iridiumTeams.getCommands().flyCommand.permission)) return true;
-        if (isBypassing()) return true;
-        Optional<T> team = iridiumTeams.getTeamManager().getTeamViaID(getTeamID());
+
+        // we should be considering if players are allowed to fly natively
+        switch(player.getGameMode()) {
+            case CREATIVE: {}
+            case SPECTATOR: { return true; }
+        }
```

Comment:
id change this to an if statement

Reply:
whats the difference? it feels nice as a switch ngl

Reply:
Doesnt really make a huge difference tbh, but if there is only one case (e.g. return true) id opt for an if statement over a switch.

```java
if (player.getGameMode() == CREATIVE || player.getGameMode() == SPECTATOR) return true;
```

Just looks better imo

Reply:
i disagree on looks, but i do like that its a one-liner, so ill do it